components/ui/Input.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FocusEventHandler, useEffect } from "react";
+
+type InputProps = {} & React.InputHTMLAttributes<HTMLInputElement>;
+
+const defaultClassName =
+  "border-1 rounded-md border-transparent bg-anti-flash-white outline-none px-3 items-center";
+
+const Input: React.FC<InputProps> = React.forwardRef<
+  HTMLInputElement,
+  InputProps
+>((props, ref) => {
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const [focused, setFocused] = React.useState(false);
+
+  let { className, ...restProps } = props;
+  className = `${defaultClassName} ${className ?? ""}`;
+  const isNumber = props.type === "number";
+
+  if (isNumber) {
+    className += " font-mono";
+  }
```

Comment:
I would maybe rather do this since its a more standard approach in our code base than mutating the classname strings directly.

```ts
  let { className, ...restProps } = props;
  const isNumber = props.type === "number";
  className = `${defaultClassName} ${className ?? ""} ${isNumber ? "font-mono" : ""}`;
```

But I dont have a big problem with it tbh. Just a suggestion.